### PR TITLE
Iter5

### PR DESCRIPTION
**Pull request by ElizavetaFirst**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,175 @@
+.DEFAULT_GOAL := build
```
Comment (arttet) [inline Makefile:1]: Please avoid duplicate code

File: cmd/agent/main.go [hunk 2]
```diff
@@ -1,3 +1,78 @@
...
+	"github.com/spf13/cobra"
+)
+
+var (
+	addr           string
+	reportInterval time.Duration
+	pollInterval   time.Duration
+)
+
+func getEnvDuration(key string, defaultVal time.Duration) time.Duration {
```
Comment (arttet) [inline cmd/agent/main.go]: Please move it to another package

File: cmd/agent/main.go [hunk 3]
```diff
@@ -1,3 +1,78 @@
...
+}
+
+func getEnvString(key, defaultVal string) string {
+	if envVal, exists := os.LookupEnv(key); exists {
+		return envVal
+	}
+	return defaultVal
+}
+
+var rootCmd = &cobra.Command{
```
Comment (arttet) [inline cmd/agent/main.go]: It shouldn't be in main.cpp

File: internal/agent/uploader/uploader.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		<-ticker.C
+		u.SendGaugeMetrics(u.gaugeMetricsFunc())
+		u.SendCounterMetrics(u.counterMetricsFunc())
+	}
+}
+
+func (u *Uploader) SendGaugeMetrics(metrics map[string]float64) {
+	client := &http.Client{}
+	for k, v := range metrics {
+		req, _ := http.NewRequest("POST", "http://"+u.addr+"/update/gauge/"+k+"/"+strconv.FormatFloat(v, 'f', -1, 64), nil)
```
Comment (arttet) [inline internal/agent/uploader/uploader.go]: Please use `format`

File: internal/server/handler/handler.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+	c.String(http.StatusOK, "%v", value.Value) // возвращаем значение
+}
+
+func (h *Handler) handleGetAllValues(c *gin.Context) {
+	values := h.Storage.GetAll()
+
+	// создаем HTML ответ
+	htmlResponse := "<html><body>"
+	for name, metric := range values {
```
Comment (arttet) [inline internal/server/handler/handler.go:83]: Please use `strings.Builder`

File: internal/server/handler/handler_test.go [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/ElizavetaFirst/go-metrics-alerts/internal/server/storage"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockStorage struct{}
```
Comment (arttet) [inline internal/server/handler/handler_test.go:13]: It should be an interface

File: internal/server/storage/storage.go [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+	Value any
+}
+
+type Storage interface {
+	Update(metricName string, update Metric) error
+	Get(metricName string) (Metric, bool)
+	GetAll() map[string]Metric
+}
+
+type MemStorage struct {
```
Comment (arttet) [inline internal/server/storage/storage.go]: It should be private

File: internal/server/storage/storage.go [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+)
+
+type MetricType string
+
+const (
+	Gauge   MetricType = "gauge"
+	Counter MetricType = "counter"
+)
+
+type Metric struct {
```
Comment (arttet) [inline internal/server/storage/storage.go]: It should be in the model package

File: internal/server/storage/storage.go [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+type Storage interface {
+	Update(metricName string, update Metric) error
+	Get(metricName string) (Metric, bool)
+	GetAll() map[string]Metric
+}
+
+type MemStorage struct {
+	Data map[string]Metric
```
Comment (arttet) [inline internal/server/storage/storage.go]: It would be better if you use `sync.map`

File: internal/server/storage/storage.go [hunk 10]
```diff
@@ -0,0 +1,68 @@
+package storage
+
+import (
```
Comment (arttet) [inline internal/server/storage/storage.go:3]: Please apply gofmt

**Review by arttet — CHANGES_REQUESTED** [review]
Please fix CI

Please read it
https://cbea.ms/git-commit/
https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/

Please add linters in CI https://github.com/golangci/golangci-lint
Please add the code coverage report in CI https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
Please add gofmt in CI https://github.com/marketplace/actions/check-code-formatting-using-gofmt

File: internal/server/handler/handler_test.go (hunk 6 above)
Comment (ElizavetaFirst) [reply internal/server/handler/handler_test.go:13]: у меня его тогда в функциях не выходит использовать
как я понимаю,нужно структура и интерфейс который она реализует,для использования

**Pull request closed without merging**